src/Integrators.jl

```diff
@@ -133,11 +200,15 @@ function evolve!(integrator::LangevinHeunP, Δt::Float64)
     field!(_B, _S₁, Z, sys.hamiltonian)
     @. _S₂ = normalize(S + 0.5 * Δt * (_f₁ + f(_S₁, _B, α)) + 0.5 * √Δt * (_r₁ + f(_S₁, _ξ, α)))
```

Comment:
Rather than writing into `_S₂` and then swapping buffers, can we actually just directly write into `S`? We could keep `SpinSystem` immutable in that case, and would save some memory.

Reply:
I think you are right here, too.  I didn't want to touch the old integrators on the first pass, but I'll see if I can set it up in a manner similar to what you describe.

Reply:
Writing directly into S is the way to go.